sound/soc/sof/loader.c

```diff
@@ -32,6 +32,33 @@ static int get_ext_windows(struct snd_sof_dev *sdev,
 	return 0;
 }
 
+static int get_cc_info(struct snd_sof_dev *sdev,
+		       struct sof_ipc_ext_data_hdr *ext_hdr)
+{
+	int ret;
+
+	struct sof_ipc_cc_version *cc =
+		container_of(ext_hdr, struct sof_ipc_cc_version, ext_hdr);
+
+	dev_dbg(sdev->dev, "Firmware info: used compiler %s %d:%d:%d%s used optimization flags %s\n",
+		 cc->name, cc->major, cc->minor, cc->micro, cc->desc,
+		 cc->optim);
+
+	/* create read-only cc_version debugfs to store compiler version info */
+	if (sdev->first_boot) {
+		ret = snd_sof_debugfs_buf_item(sdev, cc, cc->ext_hdr.hdr.size,
+					"cc_version", 0444);
+
+		/* errors are only due to memory allocation, not debugfs */
+		if (ret < 0) {
+			dev_err(sdev->dev, "error: snd_sof_debugfs_buf_item failed\n");
+			return ret;
+		}
```

Comment:
@ktrzcinx dont you need to save the cc info somewhere? the debugfs entry points to the address which will be freed after the extended data is freed no?

Reply:
Well, there is no such an information in debugfs description (and functions below) and test on real machine show that file has been written correctly

Reply:
@ktrzcinx take a look at sof_dfsentry_read(). This function tried to mopy the data from the df->buf which points to the base address that is passed when the dfsentry is created. In this case it is pointing to the ext_data that is passed to get_cc_info() which is eventually freed when the snd_sof_fw_parse_ext_data() function quits.

Reply:
Ok, done

Reply:
Ok, done